api/src/main/java/org/openmrs/api/db/hibernate/HibernateObsDAO.java

```diff
@@ -255,5 +255,15 @@ public Obs getObsByUuid(String uuid) {
 		return (Obs) sessionFactory.getCurrentSession().createQuery("from Obs o where o.uuid = :uuid").setString("uuid",
 		    uuid).uniqueResult();
 	}
-	
+
+	/**
+	 * @see org.openmrs.api.db.ObsDAO#getRevisionObs(org.openmrs.Obs)
+	 */
+	public Obs getRevisionObs(Obs initialObs) {
+		return (Obs) sessionFactory.getCurrentSession()
+				.createQuery("from Obs o where o.previousVersion = :id")
+				.setString("id",initialObs.getId().toString())
```

Comment:
Do you need to get the id? Hibernate should be able to work with the Obs object rather than its id, it can help to avoid a potential NullPointerException